en/orm/database-basics.rst

```diff
@@ -442,6 +442,114 @@ your Table's :ref:`_initializeSchema() method <saving-complex-types>`::
+--------------------------------------------
+
+.. versionadded:: 3.3.0
+    Support for mapping custom data types to SQL expressions was added in 3.3.0.
+
+The previous example maps a custom datatype for a 'json' column type which is
+easily represented as a string in a SQL statement. Complex SQL data
+types cannot be represented as strings/integers in SQL queries. When working
+with these datatypes your Type class needs to implement the
+``Cake\Database\Type\ExpressionTypeInterface`` interface. This interface lets
+your custom type represent a value as a SQL expression. As an example, we'll
+build a simple Type class for handling ``POINT`` type data out of MySQL. First
+we'll define a 'value' object that we can use to represent ``POINT`` data in
+PHP::
+
+    // in src/Database/Point.php
+    namespace App\Database;
+
+    // Our value object is immutable.
+    class Point
+    {
+        protected $_lat;
+        protected $_long;
+
+        // Factory method.
+        public static function parse($value)
+        {
+            // Parse the data from MySQL.
+            return new static($part[0], $part[1]);
+        }
+
+        public function __construct($lat, $long)
+        {
+            $this->_lat = $lat;
+            $this->_long = $long;
+        }
+
+        public function lat()
+        {
+            return $this->_lat;
+        }
+
+        public function long()
+        {
+            return $this->_long;
+        }
+    }
+
+With our value object created, we'll need a Type class to map data into this
+value object and into SQL expressions::
+
+    namespace App\Database\Type;
+
+    use App\Database\Point;
+    use Cake\Database\Expression\FunctionExpression;
+    use Cake\Database\Type as BaseType;
+    use Cake\Database\Type\ExpressionTypeInterface;
+
+    class PointType extends BaseType implements ExpressionTypeInterface
+    {
+        public function toPHP($value, Driver $d)
+        {
+            return Point::parse($value);
+        }
+
+        public function marshall($value)
+        {
+            if (is_string($value)) {
+                $value = extract(',', $value);
+            }
+            if (is_array($value)) {
+                return new Point($value[0], $value[1]);
+            }
+            return null;
+        }
+
+        public function toExpression($value)
+        {
+            if ($value instanceof Point) {
+                return new FunctionExpression(
+                    'POINT',
+                    $value->lat(),
+                    $value->long()
+                );
+            }
+            if (is_array($value)) {
+                return new FunctionExpression('POINT', $value[0], $value[1]);
+            }
+            // Handle other cases.
+        }
+    }
+
+The above class does a few interesting things:
+
+* The ``toPHP`` method handles parsing the SQL query results into a value
+  object.
+* The ``marshall`` method handles converting POST data into our value object.
```

Comment:
Not necessarily POST but just given data?

Reply:
$value = unpack('x4/corder/Ltype/dlat/dlon', $value); can be used to parse mysql binary point data in class Point:
	public static function parse($value)
	{
		// Parse the data from MySQL.
		$value = unpack('x4/corder/Ltype/dlat/dlon', $value);
		return new static($value['lat'], $value['lon']);
	}
Not sure how it can be changed in documentation